level1a/handlers/level1a.py

```diff
@@ -37,39 +45,42 @@ def get_or_raise(variable_name: str) -> str:
     return var
 
 
-def get_last_date(
-    dataset: ds.FileSystemDataset,
-    start: Timestamp,
-) -> Optional[Timestamp]:
+def parse_event_message(event: Event) -> Tuple[str, str]:
     try:
-        dates = dataset.to_table(
-            filter=ds.field('EXPDate') >= start, columns=['EXPDate']
-        )['EXPDate'].to_pandas()
-    except pa.ArrowInvalid:
-        return None
-    if len(dates) == 0:
-        return None
-    return dates.max()
+        message: Dict[str, Any] = json.loads(event["Records"][0]["body"])
+        bucket = message["Records"][0]["s3"]["bucket"]["name"]
+        key = message["Records"][0]["s3"]["object"]["key"]
+    except (KeyError, TypeError):
+        raise InvalidMessage
+    return bucket, key
+
+
+def covers(
+    indices: DatetimeIndex,
+    first: Timestamp,
+    last: Timestamp,
+) -> bool:
+    return not (
+        len(indices) == 0
+        or first < indices.min()
+        or last > indices.max()
+    )
 
 
 def get_ccd_records(
     path_or_bucket: str,
-    min_time: Timestamp,
     filesystem: pa.fs.FileSystem = None,
 ) -> DataFrame:
-    return ds.dataset(
+    return pq.read_table(
```

Comment:
Why not `read_pandas` directly?

Reply:
Because it has the same return type as `read_table` but tries to read pandas metadata as well, which shouldn't exist since the file is written in Go
https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetDataset.html#pyarrow.parquet.ParquetDataset.read_pandas

Reply:
I see, thanks!